src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoResolver.cs

```diff
@@ -23,32 +25,42 @@ public static class JsonTypeInfoResolver
         /// which typically define contract metadata for small subsets of types.
         /// It can also be used to fall back to <see cref="DefaultJsonTypeInfoResolver"/> wherever necessary.
         /// </remarks>
-        public static IJsonTypeInfoResolver Combine(params IJsonTypeInfoResolver[] resolvers)
+        public static IJsonTypeInfoResolver Combine(params IJsonTypeInfoResolver?[] resolvers)
         {
-            if (resolvers == null)
+            if (resolvers is null)
             {
                 throw new ArgumentNullException(nameof(resolvers));
             }
 
+            var flattenedResolvers = new List<IJsonTypeInfoResolver>();
+
             foreach (IJsonTypeInfoResolver? resolver in resolvers)
             {
-                if (resolver == null)
+                if (resolver is null)
+                {
+                    continue;
+                }
```

Comment:
doesn't null occurrence mean we should use fallback to reflection? That was my understanding from discussion at least

Reply:
we should clarify XML docs to mention what null does (remark or something)

Reply:
Yep, we should update API docs in both methods.